Editor/CoreAPI/StateManager.cs

```diff
@@ -16,40 +20,251 @@ public class StateManager
+                SaveProfiles();
+            }
+        }
+
+        public string ManagedEC2FleetName
+        {
+            get => _selectedProfile.ManagedEC2FleetName;
+            set
+            {
+                _selectedProfile.ManagedEC2FleetName = value;
+                SaveProfiles();
+            }
+        }
+
+        public string BuildName
+        {
+            get => _selectedProfile.BuildName;
+            set
+            {
+                _selectedProfile.BuildName = value;
+                SaveProfiles();
+            }
+        }
+
+        public string LaunchParameters
+        {
+            get => _selectedProfile.LaunchParameters;
+            set
+            {
+                _selectedProfile.LaunchParameters = value;
+                SaveProfiles();
+            }
+        }
+
+        public string BuildOperatingSystem
+        {
+            get => _selectedProfile.BuildOperatingSystem;
+            set
+            {
+                _selectedProfile.BuildOperatingSystem = value;
+                SaveProfiles();
+            }
+        }
+
+        public string DeploymentBuildFilePath
+        {
+            get => _selectedProfile.DeploymentBuildFilePath;
+            set
+            {
+                _selectedProfile.DeploymentBuildFilePath = value;
+                SaveProfiles();
+            }
+        }
+
+        public string DeploymentBuildFolderPath
+        {
+            get => _selectedProfile.DeploymentBuildFolderPath;
+            set
+            {
+                _selectedProfile.DeploymentBuildFolderPath = value;
+                SaveProfiles();
+            }
+        }
+
+        #endregion
+
+        public IReadOnlyList<string> AllProfiles => CoreApi.ListCredentialsProfiles().Profiles.ToList();
+
+        public bool IsBootstrapped => !string.IsNullOrWhiteSpace(_selectedProfile?.Name) &&
+                                      !string.IsNullOrWhiteSpace(_selectedProfile?.Region) &&
+                                      !string.IsNullOrWhiteSpace(_selectedProfile?.BucketName);
+
+        public Action OnUserProfileUpdated { get; set; }
 
         public StateManager(CoreApi coreApi)
         {
             CoreApi = coreApi;
             AmazonGameLiftWrapperFactory = new AmazonGameLiftWrapperFactory(coreApi);
-
+            RefreshProfiles();
             SetProfile(coreApi.GetSetting(SettingsKeys.CurrentProfileName).Value);
         }
 
-        private void SetProfile(string profileName)
+        public void SetProfile(string profileName)
         {
-            if (string.IsNullOrWhiteSpace(profileName) || profileName == _selectedProfile) return;
-            _selectedProfile = profileName;
+            if (string.IsNullOrWhiteSpace(profileName) || profileName == ProfileName) return;
+
+            _selectedProfile = _allProfiles.FirstOrDefault(profile => profile.Name == profileName);
+            if (_selectedProfile == null)
```

Comment:
As we are querying for an existing profile by name, this can return null if a profile has not been created in the instance that we are selecting a specific profile for the first time